Update runtime data with CMake

abannerth opened the pull request:
This will generate runtime data as part of the ALL build (or when explicitly running the update_data target); the output file is currently generated under `BUILD_DIR/run/data.pkg`.

Modifying any files in `run/data` will trigger the update of the package, though if new files are added (or something deleted), a `make rebuild_cache` is necessary.

abannerth committed:
Get rid of \303\274 in file name.

abannerth committed:
Handle absolute paths as well as relative when compressing data

abannerth committed:
Add the new 'update_data' target, to generate the runtime data package

MrPepperoni commented on run/CMakeLists.txt:
```diff
@@ -0,0 +1,18 @@
+add_custom_command(OUTPUT data.pkg
+        COMMAND
+            ${SETENV_FOR_UPDATE_DATA}
+            r2pkg -c ${CMAKE_CURRENT_SOURCE_DIR}/data
```
I'd rather use find_program to check if r2pkg is available

abannerth replied:
Using the executable target name as the command tells cmake to replace the call with the location of the executable created during build time, and also to create a target-level dependency (so r2pkg would be always built before the custom command runs).

True, this might not be the case on Windows, where the first command is the `${CMAKE_COMMAND}` and not the target name, but in that case using the `$<TARGET_FILE_DIR:r2pkg>` generator expression in the arguments has the same effect (creates a target level dependency).

Additionally, `find_program` would run during configuration time, when the r2pkg might not be built yet. The only use case I imagine for that would be if you wanted to use an independently built/installed r2pkg to run the update target

MrPepperoni replied:
Yeah, that's what i had in mind, in case of crosscompilation. Although we might as well skip the update_data step in that case.

abannerth replied:
Ok, so which would you prefer:
1. When crosscompiling, don't add update_data to ALL
2. When crosscompiling, use find_program to check if r2pkg is available (and if not, skip the update target?)

MrPepperoni replied:
2please :)

abannerth committed:
Handle update_data properly when crosscompiling

Use r2pkg executable from the build platform, if available;
otherwise don't even add the target.

abannerth commented:
Better?
